rust/src/compose.rs

```diff
@@ -312,14 +323,116 @@ impl BuildChunkedOCIOpts {
     }
 }
 
+/// Given a .repo file, rewrite all references to gpgkey=file:// inside
+/// it to point into the source_root (if the key can be found there).
+/// This is a workaround for https://github.com/coreos/rpm-ostree/issues/5285
+fn mutate_one_dnf_repo(
```

Comment:
Hmm, probably not worth rewriting, but is there a reason we don't parse this as keyfiles instead like libdnf does?

Reply:
If we parsed like that we'd lose whitespace and comments I think without more work; which I guess wouldn't matter really.

In the end the format is so simple that parsing this way I think is going to be pretty reliable.